monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/delegates/RegistryResourceDelegate.java

```diff
@@ -23,93 +23,217 @@
-            List<RegistryProperty> properties = new ArrayList<>();
-            for (JsonElement propertyElement : propertiesJson) {
-                RegistryProperty registryProperty = new RegistryProperty();
-                JsonObject propertyObject = propertyElement.getAsJsonObject();
-                String propertyName = propertyObject.get("name").getAsString();
-                String propertyValue = propertyObject.get("value").getAsString();
-                registryProperty.setPropertyName(propertyName);
-                registryProperty.setPropertyValue(propertyValue);
-                properties.add(registryProperty);
+        JsonArray registryArray = getSearchedRegistryInPath(groupId, searchKey, filePath);
+        if (searchKey.length() == 0) {
+            for (JsonElement jsonElement : registryArray) {
+                JsonObject registryObject = jsonElement.getAsJsonObject();
+                String childName = registryObject.get("name").getAsString();
+                String mediaType = registryObject.get("mediaType").getAsString();
+                
+                RegistryArtifactsInner registryArtifactsInner = new RegistryArtifactsInner();  
+                registryArtifactsInner.setChildName(childName);
+                registryArtifactsInner.setChildPath("");
+                registryArtifactsInner.setMediaType(mediaType);
+                registryList.add(registryArtifactsInner);
+            }
+        } else {
+            for (JsonElement jsonElement : registryArray) {
+                JsonObject registryObject = jsonElement.getAsJsonObject();
+                String mediaType = registryObject.get("type").getAsString();
+                String childName = registryObject.get("name").getAsString();
+                String path = registryObject.get("path").getAsString();
+                RegistryArtifactsInner registryArtifactsInner = new RegistryArtifactsInner();
+                
+                registryArtifactsInner.setChildName(childName);
+                registryArtifactsInner.setChildPath(path); //for clarity of search results
+                registryArtifactsInner.setMediaType(mediaType);
+                registryList.add(registryArtifactsInner);
             }
-            RegistryArtifactsInner registryArtifactsInner = new RegistryArtifactsInner();
-            registryArtifactsInner.setChildName(childName);
-            registryArtifactsInner.setMediaType(mediaType);
-            registryArtifactsInner.setProperties(properties);
-            registryList.add(registryArtifactsInner);
         }
+        //ordering   
+        Comparator<RegistryArtifactsInner> comparatorObject;
+        switch (orderBy.toLowerCase()) {
+            //for any other ordering options
+            default: comparatorObject = Comparator.comparing(RegistryArtifactsInner::getChildNameIgnoreCase); break;
+        }
+        if (order.equalsIgnoreCase("desc")) {
```

Comment:
Can order become null here? There are few other places too.. If it is possible let;s switch the two parameters.

Reply:
'order' is assigned a default value from the frontend, which is coming here via an internal API Call.